public/app/features/plugins/admin/components/InstallControls/InstallControls.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { css } from '@emotion/css';
+
+import { config } from '@grafana/runtime';
+import { HorizontalGroup, Icon, LinkButton, useStyles2 } from '@grafana/ui';
+import { GrafanaTheme2, PluginType } from '@grafana/data';
+
+import { ExternallyManagedButton } from './ExternallyManagedButton';
+import { InstallControlsButton } from './InstallControlsButton';
+import { CatalogPlugin, PluginStatus, Version } from '../../types';
+import { getExternalManageLink, isInstallControlsEnabled } from '../../helpers';
+import { useIsRemotePluginsAvailable } from '../../state/hooks';
+import { isGrafanaAdmin } from '../../permissions';
+
+interface Props {
+  plugin: CatalogPlugin;
+  latestCompatibleVersion?: Version;
+}
+
+export const InstallControls = ({ plugin, latestCompatibleVersion }: Props) => {
+  const styles = useStyles2(getStyles);
+  const isExternallyManaged = config.pluginAdminExternalManageEnabled;
+  const hasPermission = isGrafanaAdmin();
+  const isRemotePluginsAvailable = useIsRemotePluginsAvailable();
+  const isCompatible = Boolean(latestCompatibleVersion);
+  const isInstallControlsDisabled =
+    plugin.isCore || plugin.isDisabled || plugin.type === PluginType.renderer || !isInstallControlsEnabled();
+
+  const pluginStatus = plugin.isInstalled
+    ? plugin.hasUpdate
+      ? PluginStatus.UPDATE
+      : PluginStatus.UNINSTALL
+    : PluginStatus.INSTALL;
+
+  if (isInstallControlsDisabled) {
+    return null;
+  }
+
+  if (plugin.isEnterprise && !config.licenseInfo?.hasValidLicense) {
+    return (
+      <HorizontalGroup height="auto" align="center">
+        <span className={styles.message}>No valid Grafana Enterprise license detected.</span>
+        <LinkButton
+          href={`${getExternalManageLink(plugin.id)}?utm_source=grafana_catalog_learn_more`}
+          target="_blank"
+          rel="noopener noreferrer"
+          size="sm"
+          fill="text"
+          icon="external-link-alt"
+        >
+          Learn more
+        </LinkButton>
+      </HorizontalGroup>
+    );
+  }
+
+  if (plugin.isDev) {
+    return (
+      <div className={styles.message}>This is a development build of the plugin and can&#39;t be uninstalled.</div>
+    );
+  }
+
+  if (!hasPermission && !isExternallyManaged) {
+    const message = `You do not have permission to ${pluginStatus} this plugin.`;
+    return <div className={styles.message}>{message}</div>;
+  }
+
+  if (!plugin.isPublished) {
+    return (
+      <div className={styles.message}>
+        <Icon name="exclamation-triangle" /> This plugin is not published to{' '}
```

Comment:
Unless you think the Icon here provides extra information, in which case `alt="something"` should be added to `Icon`, `alt=""` should be added so it's ignored by screen readers I think.

Reply:
Thanks @kaydelaney, that's a good point 👍

Reply:
Nice, I didn't know this. Thanks for pointing this out @kaydelaney  🙏🏻

Reply:
@kaydelaney I hope you don't mind, but I probably will follow up on these in a separate issue as I am a bit uncertain about how to do it properly. (messaged you on Slack as well, will update this thread with the outcome of our conversation).

Issue for this: https://github.com/grafana/grafana/issues/41947